backbone/src/test/java/org/researchstack/backbone/ui/task/TaskViewModelTest.kt

```diff
@@ -0,0 +1,419 @@
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(nextStepMocked)
+        `when`(taskViewModel.isReviewStep(nextStepMocked)).thenReturn(true)
+
+        // Act
+        taskViewModel.removeUpdatedLayout()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.editing, false)
+        Assert.assertEquals(taskViewModel.updateCancelEditInLayout.value, true)
+        Assert.assertEquals(taskViewModel.hideMenuItemCancel.value, false)
+    }
+
+    @Test
+    fun testNext_ShouldUpdateCurrentStepToNextStep_shouldUpdateCurrentStepEventToNextStep() {
+        // Assemble
+        taskViewModel = createViewModel()
+        val currentStepMocked = createCurrentStep()
+        val nextStepMocked = createCurrentStep("nextStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(nextStepMocked)
+
+        // Act
+        taskViewModel.nextStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, nextStepMocked)
+        Assert.assertEquals(taskViewModel.currentStepEvent.value!!.step, nextStepMocked)
+    }
+
+    @Test
+    fun testPrevious_ShouldUpdateCurrentStepToPreviousStep_shouldUpdateCurrentStepEventToPreviousStep() {
+        // Assemble
+        taskViewModel = createViewModel()
+
+        val currentStepMocked = createCurrentStep()
+        val previousStepMocked = createCurrentStep("previousStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepBeforeStep(any(), any())).thenReturn(previousStepMocked)
+
+        // Act
+        taskViewModel.previousStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, previousStepMocked)
+        Assert.assertEquals(taskViewModel.currentStepEvent.value!!.step, previousStepMocked)
+    }
+
+    @Test
+    fun checkPrevious_shouldCloseWhenPreviousIsNull() {
+        // Assemble
+        taskViewModel = createViewModel()
+        `when`(task!!.getStepBeforeStep(any(), any())).thenReturn(null)
+        val currentStepMocked = createCurrentStep()
+        taskViewModel.currentStep = currentStepMocked
+
+        // Act
+        taskViewModel.previousStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.taskCompleted.value, false)
+    }
+
+    @Test
+    fun testNext_shouldCloseWhenNextNull() {
+        // Assemble
+        taskViewModel = spy(createViewModel())
+
+        // Act
+        taskViewModel.nextStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.taskCompleted.value, true)
+    }
+
+    @Test
+    fun testShowCurrentStep_ifCurrentStepIsNull_ShowGoToNextStep() {
+        // Assemble
+        taskViewModel = spy(createViewModel())
+        // Act
+        taskViewModel.showCurrentStep()
+
+        // Assert
+        verify(taskViewModel, times(1)).showCurrentStep()
```

Comment:
This is not currently testing that you go to the next step when the current step is null. It's only testing that the showCurrentStep() method is called. But what happens in there is not tested.

I believe you want to test that `nextStep()` is also called.

Reply:
should verify nextStep() instead of showCurrentStep() , but I remove this test, i don't found it useful